src/rest_api/classes/person/widgets/lineage.clj

```diff
@@ -0,0 +1,349 @@
+    (for [level levels
+          data (if ((keyword level) roleColour) 
+                 (vector
+                   (hash-map
+                     :personid personid
+                     :personname personname
+                     :otherpersonid otherpersonid
+                     :otherpersonname otherpersonname
+                     :level level))
+                 nil )] data)))
+
+(defn- generate-map-this-person [oid]
+  (let [personid (get oid 0)
+        personname (get oid 1)
+        data (vector
+               (hash-map
+                 :personid personid
+                 :personname personname))] data))
+
+(defn- get-this-node-scaling [queriedData]
+  (let [value (first queriedData)
+        data  (if (nil? value)
+                1
+                (if (nil? ((keyword (:personid value)) scalingHash))
+                  1 
+                  (parse-int ((keyword (:personid value)) scalingHash))))] data))
+
+(defn- get-other-node-scaling [queriedData] 
+  (if (empty? queriedData) nil
+    (for [value queriedData
+          :let [data 
+                (if (nil? value)
+                  1
+                  (if (nil? ((keyword (:otherpersonid value)) scalingHash))
+                    1 
+                    (parse-int ((keyword (:otherpersonid value)) scalingHash))))]] data)))
+
+(defn- filter-this-node [directOrFull queriedData]
+  (let [value (first queriedData)
+        scaling (if (nil? ((keyword (:personid value)) scalingHash))
+                  1 
+                  (parse-int ((keyword (:personid value)) scalingHash)))
+        data (str "id: '" directOrFull (:personid value) "', name: '" (:personname value) "', url: '" (:personid value) "', scaling: '" scaling "', radius: '100', nodeshape: 'rectangle'")]
+    data))
+
+(defn- filter-other-nodes [directOrFull largestNodeScaling queriedData]
+  (for [value queriedData
+        :let [scaling (if (nil? value) 
+                        1
+                        (if (nil? ((keyword (:otherpersonid value)) scalingHash))
+                          1 
+                          (parse-int ((keyword (:otherpersonid value)) scalingHash))))
+              radius 100
+              radius (+ 25 (* 50 (/ (Math/log scaling) (Math/log largestNodeScaling))))
+              data (if (nil? value) nil (str "id: '" directOrFull(:otherpersonid value) "', name: '" (:otherpersonname value) "', url: '" (:otherpersonid value) "', scaling: '" scaling "', radius: '" radius "', nodeshape: 'ellipse'"))]] data))
+
+(defn- get-existing-roles [queriedData]
+  (for [value queriedData
+        :let [data (str (:level value))]] data))
+
+(defn- filter-existing-roles [existingRolesUnique]
+  (apply merge (for [role existingRolesUnique
+                     :let [data  
+                           (pace-utils/vmap
+                             (keyword role) ((keyword role) roleColour))]] data)))
+
+(defn- filter-edges-supervisees [directOrFull queriedData]
+  (if (empty? queriedData)
```

Comment:
Is this the example referred to above that should be using not-empty ? I'd seen code in the format (if x nil y) and just replicated it.  I've changed it to the syntax in this example, and it seems to run well in my test cases.  Should it be using not-empty in some way instead ?